packages/workspace/src/generators/move/lib/create-project-configuration-in-new-destination.ts

```diff
@@ -35,8 +36,23 @@ export function createProjectConfigurationInNewDestination(
       `"${schema.relativeToRootDestination}/src/$2"`
     );
   } else {
+    // There's another issue if project name === path, where the target
+    // string (my-app:build:production) was being replaced
+    // and resulting in my-destination/my-new-name:build:production
+
+    // So we need to safeguard for this, and first replace the name in target strings
+    // and then replace any paths.
+
+    // Only change the target strings (my-app:build:production)
+    newProjectString = newProjectString.replace(
```

Comment:
This is needed, the old and new names might be the same anyway.

Reply:
If they are not, though?

Reply:
I look like we rename target string in subsequent functions like `updateBuildTargets`. So this would make those obsolete, worth taking a look at whether we can remove some of those logic?